swift/Samples/TabBarContainer/DemoApp/DemoWorkflow.swift

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Workflow
+import WorkflowUI
+import TabBarContainer
+
+
+// MARK: Input and Output
+
+struct DemoWorkflow: Workflow {
+
+    typealias Output = Never
+
+}
+
+
+// MARK: State and Initialization
+
+extension DemoWorkflow {
+
+    enum State {
+        case baz
+        case foo
+    }
+
+    func makeInitialState() -> State {
+        return .baz
+    }
+
+    func workflowDidChange(from previousWorkflow: DemoWorkflow, state: inout State) {
+    }
+
+}
+
+// MARK: Actions
+
+extension DemoWorkflow {
+    enum Action: WorkflowAction {
+        typealias WorkflowType = DemoWorkflow
+        
+        case bazTabSelected
+        case fooTabSelected
+        
+        func apply(toState state: inout DemoWorkflow.State) -> Never? {
+            switch self {
+            case .bazTabSelected:
+                state = .baz
+            case .fooTabSelected:
+                state = .foo
+            }
+            
+            return nil
+        }
+    }
+}
+
+
+// MARK: Rendering
+
+extension DemoWorkflow {
+
+    typealias Rendering = TabBarScreen<AnyScreen>
+
+    func render(state: State, context: RenderContext<DemoWorkflow>) -> Rendering {
+        let sink = context.makeSink(of: Action.self)
```

Comment:
This seems like a lot of code to render the tab bar view. Anything we can do to tighten this up?

Reply:
Explored a similar direction here: https://github.com/square/workflow/pull/1106/files#diff-09f70fe452e73d20c303e8566efc935bR87

Reply:
Accepted @dhavalshreyas's suggestions.